### PR TITLE
ci: onboarding module to ibm registery

### PR DESCRIPTION
**Pull request by rajatagarwal-ibm**
### Description

Onboarding module 

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: examples/basic/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Complete example
```
Comment (ocofaigh) [inline examples/basic/README.md:1]: Basic

File: examples/basic/catalogValidationValues.json.template [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "ibmcloud_api_key": $VALIDATION_APIKEY,
+  "region": "us-south",
+  "resource_tags": $TAGS,
+  "prefix": $PREFIX
+}
```
Comment (ocofaigh) [inline examples/basic/catalogValidationValues.json.template:6]: pass the vars to use existing secrets manager instance. Use `$SM_GUID` and `$SM_REGION` - I'll update pipeline to support this

File: README.md [hunk 3]
```diff
@@ -7,20 +7,18 @@
...
+  ibmcloud_api_key = "XXXXXXXXXX"
   region           = "us-south"
 }
 
 module "sm-secret-group-module" {
-  source  = "terraform-ibm-modules/secrets-manager-secret-group/ibm"
-  region                   = var.region
-  version = "latest" # Replace "latest" with a release version to lock into a specific release
+  source               = "terraform-ibm-modules/secrets-manager-secret-group/ibm"
+  version              = "latest" # Replace "latest" with a release version to lock into a specific release
```
Comment (ocofaigh) [inline README.md:21]: change `latest` to `X.X.X`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

File: .github/settings.yml [hunk 4]
```diff
@@ -22,7 +22,7 @@ repository:
 
   # Uncomment this description property
   # and update the description to the current repo description.
-  description: "Creates secrets in a Secrets Manager instance"
+  description: "Creates secret group in an IBM Secrets Manager instance."
```
Comment (SirSpidey) [inline .github/settings.yml:25]:
```suggestion
  description: "Creates a secret group in an IBM Cloud Secrets Manager instance."
```

**Comment by rajatagarwal-ibm** [comment]
@SirSpidey updated docs URL to https://cloud.ibm.com/docs/secrets-manager?topic=secrets-manager-best-practices-organize-secrets#best-practices-secret-groups

**Review by SirSpidey — APPROVED** [review]
👍 thanks.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**